Assets/HoloToolkit/Input/Scripts/InputSources/InputManagerExtensions.cs

```diff
@@ -0,0 +1,314 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+using UnityEngine.EventSystems;
+
+namespace HoloToolkit.Unity.InputModule
+{
+    public static partial class InputManagerExtensions
```

Comment:
I placed the original IInputSource events in a separate file to make them reusable but to emphasize that no changes are needed in InputManager.cs when adding new events. I didn't place the speech handling code in here as these is a particular case and I didn't see any reuse case. I believe most new input events will not require any change to these files. Specially important for third-party libraries (HoloToolkit plugins).
I made it partial because it allows adding extension methods in separate files but in the same class name. It makes no difference in compile and execution time but it's important for documentation. All methods will be found under the same branch.
Please note that there are already multiple managers in HoloToolkit that are partial. I always wondered if it was so that new functionalities could be added without an inheritance penalty.